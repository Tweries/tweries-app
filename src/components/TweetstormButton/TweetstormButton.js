import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const copy = {
  Tweet: 'Tweet'
};

function TweetstormButton({ disabled, onClick, waiting }) {
  return (
    <button
      className={classnames(
        'bg-gray-300 border border-gray-500 font-bold my-4 px-6 py-2 rounded self-center',
        {
          'tweries-background-color-blue-button': !(disabled || waiting),
          'cursor-auto': disabled || waiting
        }
      )}
      data-testid="tweet"
      disabled={disabled || waiting}
      onClick={onClick}
      type="button"
    >
      {waiting ? <FontAwesomeIcon icon={faSpinner} spin /> : copy.Tweet}
    </button>
  );
}

TweetstormButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  waiting: PropTypes.bool.isRequired
};

export default TweetstormButton;
