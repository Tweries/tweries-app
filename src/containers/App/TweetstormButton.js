import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import classnames from 'classnames';
import React from 'react';

const copy = {
  Tweet: 'Tweet'
};

// TODO: move to components
function TweetstormButton({ disabled, onClick, waiting }) {
  return (
    <button
      disabled={disabled}
      className={classnames(
        'bg-gray-300 border border-gray-500 font-bold px-4 mb-2 rounded self-center',
        {
          'bg-blue-600 text-white': !disabled,
          'cursor-auto': disabled
        }
      )}
      data-testid="tweet"
      onClick={onClick}
    >
      {waiting ? <FontAwesomeIcon icon={faSpinner} spin={true} /> : copy.Tweet}
    </button>
  );
}

export default TweetstormButton;
