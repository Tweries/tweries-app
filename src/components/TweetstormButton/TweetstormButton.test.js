import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import TweetstormButton from './TweetstormButton';

describe('<TweetstormButton />', () => {
  it('disabled', () => {
    const { getByTestId } = render(<TweetstormButton disabled />);

    expect(getByTestId('tweet')).toMatchSnapshot();
  });

  it('enabled', () => {
    const { getByTestId } = render(<TweetstormButton disabled={false} />);

    expect(getByTestId('tweet')).toMatchSnapshot();
  });

  it('waiting', () => {
    const { getByTestId } = render(
      <TweetstormButton disabled={false} waiting />
    );

    expect(getByTestId('tweet')).toMatchSnapshot();
  });

  it('onClick', () => {
    const onClick = jest.fn();

    const { getByTestId } = render(
      <TweetstormButton disabled={false} onClick={onClick} />
    );

    fireEvent.click(getByTestId('tweet'));

    expect(onClick).toBeCalled();
  });
});
