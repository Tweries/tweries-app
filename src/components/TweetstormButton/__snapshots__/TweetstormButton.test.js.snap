// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TweetstormButton /> disabled 1`] = `
<button
  class="bg-gray-300 border border-gray-500 font-bold my-4 px-6 py-2 rounded self-center cursor-auto"
  data-testid="tweet"
  disabled=""
  type="button"
>
  Tweet
</button>
`;

exports[`<TweetstormButton /> enabled 1`] = `
<button
  class="bg-gray-300 border border-gray-500 font-bold my-4 px-6 py-2 rounded self-center tweries-background-color-blue-button"
  data-testid="tweet"
  type="button"
>
  Tweet
</button>
`;

exports[`<TweetstormButton /> waiting 1`] = `
<button
  class="bg-gray-300 border border-gray-500 font-bold my-4 px-6 py-2 rounded self-center cursor-auto"
  data-testid="tweet"
  disabled=""
  type="button"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-spinner fa-w-16 fa-spin "
    data-icon="spinner"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
      fill="currentColor"
    />
  </svg>
</button>
`;
